test/e2e/tests/infra/infra.go

```diff
@@ -0,0 +1,248 @@
+			Type: to.StringPtr("Microsoft.Network/routeTables/providers/roleAssignments"),
+			RoleAssignmentPropertiesWithScope: &mgmtauthorization.RoleAssignmentPropertiesWithScope{
+				Scope:            to.StringPtr(fmt.Sprintf("[resourceId('Microsoft.Network/routeTables', '%s')]", "ci-routetable")),
+				RoleDefinitionID: to.StringPtr("[subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'b24988ac-6180-42a0-ab88-20f7382dd24c')]"),
+				PrincipalID:      to.StringPtr(spID),
+				PrincipalType:    mgmtauthorization.ServicePrincipal,
+			},
+		},
+		APIVersion: azureclient.APIVersions["Microsoft.Authorization"],
+		DependsOn:  []string{"dev-vnet"},
+	}
+
+	i.template.Resources = append(i.template.Resources, &rt)
+	return i
+}
+
+func (i *Infrastructure) AddCluster(spID string, spSecret string) *Infrastructure {
+
+	cluster := arm.Resource{
+		APIVersion: "2020-04-30",
+		Resource: &mgmtopenshift.OpenShiftCluster{
+			Name:     "ci-cluster",
+			Location: i.conf.Location,
+			Type:     "Microsoft.RedHatOpenShift/openShiftClusters",
+			Properties: mgmtopenshift.OpenShiftClusterProperties{
+				ClusterProfile: mgmtopenshift.ClusterProfile{
+					PullSecret:      "",
+					Domain:          "ci-infra",
+					ResourceGroupID: fmt.Sprintf("/subscriptions/%s/resourceGroups/%s", i.conf.AzureSubscriptionID, i.conf.AROResourceGroup),
+				},
+				ServicePrincipalProfile: mgmtopenshift.ServicePrincipalProfile{
+					ClientID:     spID,
+					ClientSecret: spSecret,
+				},
+				NetworkProfile: mgmtopenshift.NetworkProfile{
+					PodCIDR:     "10.128.0.0/14",
+					ServiceCIDR: "172.30.0.0/16",
+				},
+				MasterProfile: mgmtopenshift.MasterProfile{
+					VMSize:   mgmtopenshift.VMSizeStandardD8sV3,
+					SubnetID: fmt.Sprintf("/subscriptions/225e02bc-43d0-43d1-a01a-17e584a4ef69/resourceGroups/%s/providers/Microsoft.Network/virtualNetworks/dev-vnet/subnets/master-subnet", i.conf.AROResourceGroup),
+				},
+				WorkerProfiles: []mgmtopenshift.WorkerProfile{
+					{
+						Name:       "worker",
+						VMSize:     mgmtopenshift.VMSizeStandardD4sV3,
+						DiskSizeGB: 128,
+						SubnetID:   fmt.Sprintf("/subscriptions/225e02bc-43d0-43d1-a01a-17e584a4ef69/resourceGroups/%s/providers/Microsoft.Network/virtualNetworks/dev-vnet/subnets/worker-subnet", i.conf.AROResourceGroup),
+						Count:      3,
+					},
+				},
+				APIServerProfile: mgmtopenshift.APIServerProfile{
+					Visibility: mgmtopenshift.VisibilityPublic,
+				},
+				IngressProfiles: []mgmtopenshift.IngressProfile{
+					{
+						Name:       "default",
+						Visibility: mgmtopenshift.VisibilityPublic,
+					},
+				},
+			},
+		},
+		DependsOn: []string{"dev-vnet"},
+	}
+
+	i.template.Resources = append(i.template.Resources, &cluster)
+	return i
+}
+
+// Deploy puts the cloud infra in place using ARM template and Deploy method
+func (i *Infrastructure) Deploy(ctx context.Context) error {
```

Comment:
`NewInfrastructure` and `Deploy` is a bit confusing. Need to make it easier to consume for other developers. Like adding new resources. I think it can be simplified.